jupyterlab_caip_optimizer/src/components/snackbar_widget.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { Snackbar as MaterialSnackbar } from '@material-ui/core';
+import { Alert as MuiAlert, AlertProps } from '@material-ui/lab';
+import { SnackbarState, snackbarSlice } from '../store/snackbar';
+import { connect, Provider } from 'react-redux';
+import { RootState, AppDispatch } from '../store/store';
+import { ReactWidget } from '@jupyterlab/apputils';
+import { Store } from 'redux';
+
+const mapStateToProps = (state: RootState) => ({
+  ...state.snackbar,
+});
+
+const mapDispatchToProps = (dispatch: AppDispatch) => ({
+  close: () => dispatch(snackbarSlice.actions.close()),
+});
+
+type Props = SnackbarState & { close: () => void };
+
+function Alert(props: AlertProps) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+export const SnackbarUnwrapped = ({
+  open,
+  message,
+  severity,
+  close,
+}: Props) => {
+  const handleClose = (event?: React.SyntheticEvent, reason?: string) => {
+    if (reason === 'clickaway') {
```

Comment:
What does `'clickaway'` refer to? Does it mean user clicks on somewhere outside the snackbox?

Reply:
This was from the [snackbar demo on material ui](https://material-ui.com/components/snackbars/#customized-snackbars), but to my knowledge `clickaway` refers to when a user clicks off of the message to any other element on the page.